tetris.py

```diff
@@ -127,10 +127,27 @@ def new_block(self):
             "I": [(x - 2, y), (x - 1, y), (x, y), (x + 1, y)],
         }
         if self.upcoming_block is None:
-            self.current_block = random.choice(list(blocks.values()))
+            self.current_block_shape = random.choice(list(self.blocks.keys()))
+            self.current_block_center = (x, y)
         else:
             self.current_block = self.upcoming_block
-        self.upcoming_block = random.choice(list(blocks.values()))
+        self.upcoming_block = random.choice(list(self.blocks.keys()))
```

Comment:
```suggestion
        self.upcoming_block_shape = random.choice(list(self.blocks.keys()))
```

Reply:
The suggestion were part of the new push, I made today, sorry for that